lib/java/js-websocket/src/main/java/org/enso/polyfill/Main.java

```diff
@@ -0,0 +1,65 @@
+package org.enso.polyfill;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+
+import org.enso.polyfill.websocket.WebSocketPolyfill;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.HostAccess;
+import org.graalvm.polyglot.Source;
+import org.graalvm.polyglot.io.IOAccess;
+
+public class Main {
+
+    private static final String YDOC_SERVER_PATH = "/ydoc-server-bundle.js";
+    private static final String WASM_PATH = "/153299079965dcf860b5.wasm";
+
+    private Main() {
+    }
+
+    public static void main(String[] args) throws Exception {
+        ClasspathResource.createTempFile(WASM_PATH);
```

Comment:
According to @kazcw we should just call the [Parser.parseInput](https://github.com/enso-org/enso/blob/f82e8020fe077815f3a4b4bd10da73835f8847b9/lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java#L65) instead of trying to load Rust part of the parser as WASM.

Reply:
Right now versioning isn't an issue (everything is packaged as a single deliverable, distribution). But should IDE _inject its version of the yserver_ into (**completely**) different version of an engine_, then relying on engine's parser isn't an option. I should collect these constraints somewhere.

Reply:
> Right now versioning isn't an issue (everything is packaged as a single deliverable, distribution).

Versioning isn't an issue in production, but the only reason versioning doesn't currently seem like an issue during development is that the frontend and backend are loosely-coupled in practice, with our current source-code/span-tree interface. When we move the parser into the ydoc-server, it will become essential that the frontend and backend are using the same parser version (including both the WASM and `app/gui2/shared`).

Reply:
> Versioning isn't an issue in production

..in **current production** we provide two kinds of deliverables:
- [ standalone packages containing both GUI and the backend](https://github.com/enso-org/enso/releases/tag/2024.1.1-nightly.2024.3.26)
- the cloud which also packages GUI and backend of the _same release tag_ like [2024.1.1-nightly.2024.3.26](https://github.com/enso-org/enso/releases/tag/2024.1.1-nightly.2024.3.26) (right @PabloBuchu?)

With _such a setup_ we may pay no attention to versioning...

> frontend and backend are loosely-coupled in practice

... as frontend and backend are _build time coupled_ - e.g. they are build, tested and **known to work together**. In such a setup it is clear that ...

> the frontend and backend are using the same parser version 

They do use the same version, as they are built, tested, packaged together.

As such [I am suggesting](https://github.com/enso-org/enso/pull/9528#discussion_r1537323915) to @4e6 to call the JNI version of the parser. I believe it is a simpler, few lines code change. Running our parser in its _WASM version is also possible_ (there is [GraalWasm](https://github.com/oracle/graal/blob/master/wasm/README.md), but it is more complex to setup and it'd be (slightly) slower, slower to get up to the full speed and more memory hungry because of WASM interpreter metadata overhead. That's why I'd _start with integrating the parser via its JNI version_. Only then investigate how to integrate the WASM variant.

**TBD**: add a link to _incremental compilation_ plan.
**TBD**: add a link to _IDE supplies own language server & `Standard.Visualization`_ plan.

Reply:
> > Versioning isn't an issue in production
> 
> ..in **current production** we provide two kinds of deliverables:
> 
>     * [ standalone packages containing both GUI and the backend](https://github.com/enso-org/enso/releases/tag/2024.1.1-nightly.2024.3.26)
> 
>     * the cloud which also packages GUI and backend of the _same release tag_ like [2024.1.1-nightly.2024.3.26](https://github.com/enso-org/enso/releases/tag/2024.1.1-nightly.2024.3.26) (right @PabloBuchu?)
> 
> 
> With _such a setup_ we may pay no attention to versioning...
> 
> > frontend and backend are loosely-coupled in practice
> 
> ... as frontend and backend are _build time coupled_ - e.g. they are build, tested and **known to work together**. In such a setup it is clear that ...
> 
> > the frontend and backend are using the same parser version
> 
> They do use the same version, as they are built, tested, packaged together.

Right... "versioning isn't an issue in production."

However, this change would cause some inconvenience during development. Developers frequently use versions other than the packaged versions enumerated above. It is currently possible for frontend developers to pull changes from `develop` without rebuilding the backend. I do so frequently, and I expect I am not alone in this. If we make this change, the coupling will be much tighter in practice. Frontend developers (and library developers?) will need to be aware when a commit touches the parser and they need to rebuild the backend. Developers working on the parser will have a much longer code/test cycle. Maybe this is acceptable, but it's a cost we should be aware of if we're making this change.